app/controllers/v3/service_credential_bindings_controller.rb

```diff
@@ -26,14 +27,32 @@ def index
   end
 
   def show
-    message = ServiceCredentialBindingsShowMessage.from_params(query_params)
+    message = ServiceCredentialBindingShowMessage.from_params(query_params)
     invalid_param!(message.errors.full_messages) unless message.valid?
 
     ensure_service_credential_binding_is_accessible!
 
     render status: :ok, json: serialized(message)
   end
 
+  def create
+    message = ServiceCredentialBindingCreateMessage.new(hashed_params[:body])
+    unprocessable!(message.errors.full_messages) unless message.valid?
+
+    service_instance = VCAP::CloudController::ServiceInstance.first(guid: message.service_instance_guid)
+    unprocessable!("Service instance '#{message.service_instance_guid}' not found") unless !service_instance.blank? && can_read_from_space?(service_instance.space)
```

Comment:
Isn't this ths same as `unless service_instance && can_read_from_space?(service_instance.space)`

Reply:
For this particular line, yes. It's a bit safer to call the method though (we might change how we fetch the service instance in following stories [although unlikely :P]). I will change it from `!blank?` to `present?` though